testdata/p4_16_samples/psa-portid-using-newtype2.p4

```diff
@@ -0,0 +1,293 @@
+
+
+header ethernet_t {
+    bit<48> dstAddr;
+    bit<48> srcAddr;
+    bit<16> etherType;
+}
+
+header ipv4_t {
+    bit<4>  version;
+    bit<4>  ihl;
+    bit<8>  diffserv;
+    bit<16> totalLen;
+    bit<16> identification;
+    bit<3>  flags;
+    bit<13> fragOffset;
+    bit<8>  ttl;
+    bit<8>  protocol;
+    bit<16> hdrChecksum;
+    bit<32> srcAddr;
+    bit<32> dstAddr;
+}
+
+
+// From fabric.p4
+@controller_header("packet_in")
+header packet_in_header_t {
+    // Translation will occur in agent software for this field,
+    // because it is inside of a controller_header, and because
+    // PortIdInHeader_t is defined as a newtype with a
+    // @p4runtime_translate("port") annotation.  No such annotation is
+    // needed here.
+    PortIdInHeader_t ingress_port;
+}
+
+@controller_header("packet_out")
+header packet_out_header_t {
+    // No annotation needed here to get agent translation of
+    // egress_port, for same reason as in header packet_in_header_t
+    // above.
+    PortIdInHeader_t egress_port;
+}
+
+
+struct parsed_headers_t {
+    packet_in_header_t packet_in;
+    packet_out_header_t packet_out;
+    ethernet_t ethernet;
+    ipv4_t ipv4;
+}
+
+struct fabric_metadata_t {
+}
+
+action drop() {
+    mark_to_drop();
+}
+
+action nop() {
+}
+
+parser FabricParser(packet_in packet,
+                    out parsed_headers_t hdr,
+                    inout fabric_metadata_t meta,
+                    inout standard_metadata_t standard_metadata)
+{
+    state start {
+        transition accept;
+    }
+}
+
+control PacketIoIngress(inout parsed_headers_t hdr,
+                        inout fabric_metadata_t fabric_metadata,
+                        inout standard_metadata_t standard_metadata)
+{
+    apply {
+        if (hdr.packet_out.isValid()) {
+            // I know what I'm doing, I requested translation on
+            // packet_out.egress_port
+            standard_metadata.egress_spec =
+                PSA_PORT_HEADER_TO_ID(hdr.packet_out.egress_port);
+            hdr.packet_out.setInvalid();
+            exit;
+        }
+    }
+}
+
+control PacketIoEgress(inout parsed_headers_t hdr,
+                       inout fabric_metadata_t fabric_metadata,
+                       inout standard_metadata_t standard_metadata)
+{
+    apply {
+        // comparison of PortId_t to PortId_t is valid
+        if (standard_metadata.egress_port == PSA_CPU_PORT) {
+            hdr.packet_in.setValid();
+            // I know what I'm doing, I requested translation on
+            // packet_in.ingress_port
+            hdr.packet_in.ingress_port =
+                PSA_PORT_ID_TO_HEADER(standard_metadata.ingress_port);
+        }
+    }
+}
+
+control Filtering (inout parsed_headers_t hdr,
+                   inout fabric_metadata_t fabric_metadata,
+                   inout standard_metadata_t standard_metadata)
+{
+    table t {
+        key = {
+            // translated by P4Runtime agent because of annotation on PortId_t
```

Comment:
The comment is totally correct, but that's precisely the point: translated or not, P4 tables do not care.